gremlin-test/src/main/java/org/apache/tinkerpop/gremlin/process/computer/bulkdumping/BulkExportVertexProgramTest.java

```diff
@@ -0,0 +1,99 @@
+ * under the License.
+ */
+package org.apache.tinkerpop.gremlin.process.computer.bulkdumping;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.tinkerpop.gremlin.LoadGraphWith;
+import org.apache.tinkerpop.gremlin.TestHelper;
+import org.apache.tinkerpop.gremlin.process.AbstractGremlinProcessTest;
+import org.apache.tinkerpop.gremlin.process.computer.GraphComputer;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.apache.tinkerpop.gremlin.structure.io.script.ScriptResourceAccess;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.apache.tinkerpop.gremlin.LoadGraphWith.GraphData.MODERN;
+import static org.apache.tinkerpop.gremlin.structure.Column.keys;
+import static org.apache.tinkerpop.gremlin.structure.Column.values;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Daniel Kuppitz (http://gremlin.guru)
+ */
+public class BulkExportVertexProgramTest extends AbstractGremlinProcessTest {
+
+    @Test
+    @LoadGraphWith(MODERN)
+    public void shouldExportCsvFile() throws Exception {
+        if (graphProvider.getGraphComputer(graph).features().supportsResultGraphPersistCombination(GraphComputer.ResultGraph.NEW, GraphComputer.Persist.EDGES)) {
+            g.V().hasLabel("person").match(
+                    __.as("person").values("name").as("name"),
+                    __.as("person").values("age").as("age"),
+                    __.as("person").outE("created").count().as("projects")
+            ).select("person").
+                    coalesce(__.out("created"), __.identity()).
+                    coalesce(__.in("created"), __.identity()).
+                    groupCount("m").by(__.select("person", "name", "age", "projects")).cap("m").unfold().as("kv").
+                    select(keys).select("person").as("person").
+                    select("kv").select(keys).select("name").as("name").
+                    select("kv").select(keys).select("age").as("age").
+                    select("kv").select(keys).select("projects").as("projects").
+                    select("kv").select(values).map(__.union(__.identity(), __.constant(-1)).sum()).as("coworkers").
+                    select("person", "name", "age", "projects", "coworkers").
+                    program(BulkDumperVertexProgram.build().create(graph)).iterate();
+
+            final Configuration config = graph.configuration();
+            final String outputLocation = config.getString("gremlin.hadoop.outputLocation");
+            final String scriptFileName =
+                    TestHelper.generateTempFileFromResource(ScriptResourceAccess.class, "script-csv-export.groovy", "").
+                            getAbsolutePath().replace('\\', '/');
+
+            config.setProperty("gremlin.hadoop.inputLocation", outputLocation + "/~g");
```

Comment:
Mainly working around [TINKERPOP-1304](https://issues.apache.org/jira/browse/TINKERPOP-1304). But even if 1304 would be solved, I would have to set the `graphWriter` to `ScriptOutputFormat`. I will add some appropriate methods to `HadoopGraphProvider`.